temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java

```diff
@@ -232,27 +235,45 @@ public void run() {
           suspender.await();
         }
 
-        if (pollExecutor.isShutdown()) {
+        if (shouldTerminate()) {
           return;
         }
+
         task.run();
         pollBackoffThrottler.success();
+      } catch (InterruptedException e) {
```

Comment:
So, you just made `LocalActivityPollTask` consume `InterruptedException` and return immediately without exception. And this clause would only handle exceptions from `pollBackoffThrottler` and `suspender`, right?

Reply:
Makes sense.